### PR TITLE
Update getting started for ehrql

### PR DESCRIPTION
**Pull request by inglesp**

### CODE REVIEW

File: docs/getting-started.md [hunk 1]
```diff
@@ -299,8 +299,8 @@ account, for developing your own study:
        system tray docker icon; select "settings"; select "shared drives".  This
        setting does not exist in the *WSL 2* backend.
     1. Test Docker and opensafely work together. Open an Anaconda Prompt, and run
-       `opensafely pull cohortextractor`. This will pull down the OpenSAFELY
-       cohortextractor images, which can be used to run actions in your study.  The
+       `opensafely pull ehrql`. This will pull down the OpenSAFELY
+       ehrql images, which can be used to run actions in your study.  The
```
Comment (StevenMaude) [inline docs/getting-started.md:303]: thought: Unrelated to this PR, is "can be used to run actions in your study" correct here? It doesn't sound quite right to me, but could be my reading.

File: docs/getting-started.md [hunk 2]
```diff
@@ -442,47 +444,40 @@ automatically save the edits to files that you make.**
...
 1. The "Explorer" on the left hand side lists the files and folders in
-   your research repository. Find and click on the `study_definition.py`
-   file inside the `analysis` folder. This file specifies the population
-   that you'd like to study (dataset rows) and what you need to know
-   about them (dataset columns).
+   your research repository. Find and click on the `dataset_definition.py`
+   file inside the `analysis` folder. This file contains a dataset definition,
+   specifying the population that you'd like to study (dataset rows)
+   and what you need to know about them (dataset columns).
+   It is written in [ehrQL](/ehrql/).
```
Comment (StevenMaude) [inline docs/getting-started.md:451]: suggestion:

```suggestion
   It is written in [ehrQL](/ehrql/); OpenSAFELY's electronic health record query language.
```

I think it's the first time ehrQL is referred to, so worth clarifying for the reader.

File: docs/getting-started.md [hunk 3]
```diff
@@ -442,47 +444,40 @@ automatically save the edits to files that you make.**
...
-
-    age=patients.age_as_of(
-        "2019-09-01",
-        return_expectations={
-            "rate": "universal",
-            "int": {"distribution": "population_ages"},
-        },
-    ),
-)
+```python linenums="1" hl_lines="15"
```
Comment (StevenMaude) [inline docs/getting-started.md:453]: thought: Unrelated to this PR, this highlights to me that dataset definitions outside of the ehrQL repository won't be tested. Maybe we need an issue for that?

And I think we'll need to add the extra Pygments configuration to the main documentation `mkdocs.yml` so the formatting works.

File: docs/getting-started.md [hunk 4]
```diff
@@ -442,47 +444,40 @@ automatically save the edits to files that you make.**
...
+from ehrql import Dataset
+from ehrql.tables.beta.tpp import patients, practice_registrations
+
+dataset = Dataset()
```
Comment (StevenMaude) [inline docs/getting-started.md:457]: suggestion: Probably better to do this as we instruct elsewhere:

```suggestion
from ehrql import create_dataset
from ehrql.tables.beta.tpp import patients, practice_registrations

dataset = create_dataset()
```

File: docs/getting-started.md [hunk 5]
```diff
@@ -578,57 +569,59 @@ This code reads the CSV of patient data, and saves a histogram of ages to a new
 
 === "Python"
 
-    ```yaml linenums="1" hl_lines="13 14 15 16 17 18 19 20"
+    ```yaml linenums="1" hl_lines="14 15 16 17 18 19"
     version: "3.0"
 
+    # Ignore this`expectation` block. It is required but not used, and will be removed in future versions.
```
Comment (StevenMaude) [inline docs/getting-started.md:575]:
```suggestion
    # Ignore this `expectations` block. It is required but not used, and will be removed in future versions.
```

File: docs/getting-started.md [hunk 6]
```diff
@@ -782,11 +775,8 @@ detail on the subjects covered in this tutorial. For example:
 
 * There is a more complete [guide to the OpenSAFELY command-line
   tool](opensafely-cli.md).
-* The [full study definition reference](study-def.md) describes all the
-  different ways to define new variables in your study definition.
-    * *Try adding a new variable such as [Sex](study-def-variables.md#cohortextractor.patients.sex)
-    or [Region](study-def-variables.md#cohortextractor.patients.registered_practice_as_of) to the
-    study definition you created using this guide, then plot some new charts!*
+* The [ehrQL documentation](/ehrql/) contains a tutorial for ehrQL,
+  as well as a complete [schema reference](http://localhost:8000/ehrql/reference/schemas/).
```
Comment (StevenMaude) [inline docs/getting-started.md:779]: fix: Probably more like:

```suggestion
  as well as a complete [schema reference](../ehrql/reference/schemas/).
```

File: docs/getting-started.md [hunk 7]
```diff
@@ -302,8 +299,8 @@ account, for developing your own study:
        system tray docker icon; select "settings"; select "shared drives".  This
        setting does not exist in the *WSL 2* backend.
     1. Test Docker and opensafely work together. Open an Anaconda Prompt, and run
-       `opensafely pull cohortextractor`. This will pull down the OpenSAFELY
-       cohortextractor images, which can be used to run actions in your study.  The
+       `opensafely pull ehrql`. This will pull down the OpenSAFELY
+       ehrql images, which can be used to run actions in your study.  The
```
Comment (StevenMaude) [inline docs/getting-started.md:303]: nit: Not introduced by this PR, but we could fix it here:

```suggestion
       ehrql image, which can be used to run actions in your study.  The
```

**Review by StevenMaude — APPROVED** [review]

**Comment by StevenMaude** [comment]
Looks fine to me; I've added a few suggestions and thoughts. These shouldn't take long to address.

File: docs/getting-started.md (hunk 4 above)
Comment (inglesp) [reply docs/getting-started.md:457]: Ah yes, good point.  I'll update the template too.

File: docs/getting-started.md (hunk 3 above)
Comment (StevenMaude) [reply docs/getting-started.md:453]: https://github.com/opensafely/documentation/pull/1372
https://github.com/opensafely/documentation/issues/1373

**Pull request merged**